File: backend/src/main/java/com/declaration/entity/PaymentRemittance.java
package com.declaration.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 出款水单实体类（独立管理，支持与申报单多对多关联）
 */
@Data
@TableName("payment_remittance")
public class PaymentRemittance {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /** 关联申报单ID（保留兼容，新流程使用关联表） */
    private Long formId;

    /** 出款水单编号 */
    private String paymentNo;

    /** 收款人 */
    private String payeeName;

    /** 出款日期 */
    private LocalDate paymentDate;

    /** 出款金额 */
    private BigDecimal paymentAmount;

    /** 币种 */
    private String currency;

    /** 银行账户ID */
    private Long bankAccountId;

    /** 银行名称 */
    private String bankAccountName;

    /** 凭证照片URL */
    private String photoUrl;

    /** 备注 */
    private String remarks;

    /** 状态: 0-草稿 1-待审核 2-已审核 3-已驳回 */
    private Integer status;

    /** 审核备注 */
    private String auditRemark;

    /** 审核人ID */
    private Long auditBy;

    /** 审核人姓名 */
    private String auditByName;

    /** 审核时间 */
    private LocalDateTime auditTime;

    /** Flowable流程实例ID */
    private String processInstanceId;

    /** 提交时间 */
    private LocalDateTime submitTime;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableField(fill = FieldFill.INSERT)
    private Long createBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updateBy;

    /** 已关联金额合计（非数据库字段，查询时动态计算） */
    @TableField(exist = false)
    private BigDecimal totalRelatedAmount;
}

File: frontend/src/api/business/paymentRemittance.ts
import request from '@/utils/request'

/**
 * 出款水单管理 API
 */

// 出款水单类型
export interface PaymentRemittance {
  id?: number
  formId?: number
  paymentNo?: string
  payeeName?: string
  paymentDate?: string
  paymentAmount?: number
  currency?: string
  bankAccountId?: number
  bankAccountName?: string
  photoUrl?: string
  remarks?: string
  status?: number // 0-草稿 1-待审核 2-已审核 3-已驳回
  auditRemark?: string
  auditBy?: number
  auditByName?: string
  auditTime?: string
  submitTime?: string
  createTime?: string
  updateTime?: string
  totalRelatedAmount?: number
}

// 出款水单查询参数
export interface PaymentRemittanceQueryParams {
  current: number
  size: number
  status?: number
  paymentNo?: string
  relationStatus?: string // UNRELATED-未关联 RELATED-已关联 PARTIAL-未完全关联
}

// 出款水单关联申报单
export interface PaymentRemittanceFormRelation {
  relationId?: number
  formId: number
  formNo?: string
  relationType?: number
  relationAmount?: number
  createTime?: string
}

/** 获取出款水单列表(分页) */
export function getPaymentRemittanceList(params: PaymentRemittanceQueryParams) {
  return request({
    url: '/v1/payment-remittances',
    method: 'get',
    params
  })
}

/** 获取出款水单详情 */
export function getPaymentRemittanceDetail(id: number) {
  return request({
    url: `/v1/payment-remittances/${id}`,
    method: 'get'
  })
}

/** 创建出款水单 */
export function createPaymentRemittance(data: Partial<PaymentRemittance>) {
  return request({
    url: '/v1/payment-remittances',
    method: 'post',
    data
  })
}

/** 更新出款水单(草稿状态) */
export function updatePaymentRemittance(id: number, data: Partial<PaymentRemittance>) {
  return request({
    url: `/v1/payment-remittances/${id}`,
    method: 'put',
    data
  })
}

/** 删除出款水单(草稿状态) */
export function deletePaymentRemittance(id: number) {
  return request({
    url: `/v1/payment-remittances/${id}`,
    method: 'delete'
  })
}

/** 提交出款水单审核 */
export function submitPaymentRemittanceAudit(id: number) {
  return request({
    url: `/v1/payment-remittances/${id}/submit`,
    method: 'post'
  })
}

/** 审核出款水单 */
export function auditPaymentRemittance(id: number, data: {
  approved: boolean
  bankAccountId?: number
  auditRemark?: string
}) {
  return request({
    url: `/v1/payment-remittances/${id}/audit`,
    method: 'post',
    params: data
  })
}

/** 关联申报单 */
export function relateToForm(remittanceId: number, formId: number, amount?: number, relationType?: number) {
  return request({
    url: `/v1/payment-remittances/${remittanceId}/relate-form`,
    method: 'post',
    params: { formId, amount, relationType }
  })
}

/** 取消关联申报单 */
export function unrelateFromForm(remittanceId: number, formId: number) {
  return request({
    url: `/v1/payment-remittances/${remittanceId}/unrelate-form`,
    method: 'delete',
    params: { formId }
  })
}

/** 获取出款水单关联的所有申报单 */
export function getRelatedForms(remittanceId: number) {
  return request({
    url: `/v1/payment-remittances/${remittanceId}/related-forms`,
    method: 'get'
  })
}

/** 获取申报单关联的所有出款水单 */
export function getPaymentRemittancesByFormId(formId: number) {
  return request({
    url: `/v1/payment-remittances/form/${formId}`,
    method: 'get'
  })
}

/** 反审核出款水单 */
export function revokePaymentRemittanceAudit(id: number) {
  return request({
    url: `/v1/payment-remittances/${id}/revoke-audit`,
    method: 'post'
  })
}
